src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -126,16 +132,28 @@ async def _handle_unary_unary_rpc(object method_handler,
     )
 
     # Sends response message
-    cdef tuple send_ops = (
-        SendStatusFromServerOperation(
-            tuple(),
-            StatusCode.ok,
-            b'',
-            _EMPTY_FLAGS,
-        ),
-        SendInitialMetadataOperation(None, _EMPTY_FLAGS),
-        SendMessageOperation(response_raw, _EMPTY_FLAGS),
-    )
+    cdef tuple send_ops
+    if servicer_context._metadata_sent:
+        send_ops = (
+            SendStatusFromServerOperation(
+                rpc_state.trailing_metadata,
+                StatusCode.ok,
+                b'',
+                _EMPTY_FLAGS,
+            ),
+            SendMessageOperation(response_raw, _EMPTY_FLAGS),
+        )
+    else:
+        send_ops = (
+            SendStatusFromServerOperation(
+                rpc_state.trailing_metadata,
+                StatusCode.ok,
+                b'',
+                _EMPTY_FLAGS,
+            ),
+            SendInitialMetadataOperation(None, _EMPTY_FLAGS),
```

Comment:
Core may sort these operations into the proper order, but it seems somewhat unintuitive that this tuple isn't in chronological order. Can we move the `SendInitialMetadataOperation` to the top of the list?